# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsIFrameTraversal.h
# Commit: d4411b38cae7
# Full Hash: d4411b38cae7216ffc57a9b8b246806acf0a3717
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2023-12-24 09:09:23
# Regressor Bug: 779684
# File Overlap Count: 1
# Description:
#   Bug 779684 - part 1: Get rid of `nsIFrameEnumerator` interface r=emilio
#   
#   It's inherited only by `nsFrameIterator`, and `nsFrameIterator` can be declared
#   in a header file. So, the interface is not required and removing it can avoid
#   virtual calls.
# ==============================================================================

diff -r bda7c9e66417 -r d4411b38cae7 layout/base/nsIFrameTraversal.h
--- a/layout/base/nsIFrameTraversal.h	Sat Dec 23 19:02:04 2023 +0000
+++ b/layout/base/nsIFrameTraversal.h	Sun Dec 24 06:01:19 2023 +0000
@@ -8,40 +8,10 @@
 
 #include "nsISupports.h"
 
+class nsFrameIterator;
 class nsIFrame;
 class nsPresContext;
 
-#define NS_IFRAMEENUMERATOR_IID                      \
-  {                                                  \
-    0x7c633f5d, 0x91eb, 0x494e, {                    \
-      0xa1, 0x40, 0x17, 0x46, 0x17, 0x4c, 0x23, 0xd3 \
-    }                                                \
-  }
-
-class nsIFrameEnumerator : public nsISupports {
- public:
-  NS_DECLARE_STATIC_IID_ACCESSOR(NS_IFRAMEENUMERATOR_IID)
-
-  virtual void First() = 0;
-  virtual void Next() = 0;
-  virtual nsIFrame* CurrentItem() = 0;
-  virtual bool IsDone() = 0;
-
-  virtual void Last() = 0;
-  virtual void Prev() = 0;
-
-  inline nsIFrame* Traverse(bool aForward) {
-    if (aForward) {
-      Next();
-    } else {
-      Prev();
-    }
-    return CurrentItem();
-  };
-};
-
-NS_DEFINE_STATIC_IID_ACCESSOR(nsIFrameEnumerator, NS_IFRAMEENUMERATOR_IID)
-
 enum nsIteratorType { eLeaf, ePreOrder, ePostOrder };
 
 // {d33fe76c-207c-4359-a315-8eb1eecf80e5}
@@ -75,7 +45,7 @@
    * @param aLimiter [in] if this is non-null and an ancestor of aStart,
    *        iteration will be limited to just the descendants of this frame.
    */
-  NS_IMETHOD NewFrameTraversal(nsIFrameEnumerator** aEnumerator,
+  NS_IMETHOD NewFrameTraversal(nsFrameIterator** aEnumerator,
                                nsPresContext* aPresContext, nsIFrame* aStart,
                                int32_t aType, bool aVisual,
                                bool aLockInScrollView, bool aFollowOOFs,